alf/algorithms/agent.py

```diff
@@ -155,9 +175,24 @@ def rollout(self, time_step: ActionTimeStep, state: AgentState, mode):
         new_state = AgentState()
         info = AgentInfo()
         observation = self._encode(time_step)
+
+        if self._goal_generator is not None:
+            goal_step = self._goal_generator.predict(
+                observation=observation,
+                state=state.goal_generator,
+                step_type=time_step.step_type)
+            info = info._replace(goal_generator=goal_step.info)
+            new_state = new_state._replace(goal_generator=goal_step.state)
+
+            observation = [observation, goal_step.outputs]
+
         if self._icm is not None:
-            icm_step = self._icm.train_step(
-                (observation, time_step.prev_action), state=state.icm)
+            if self._goal_generator is not None:
+                icm_step = self._icm.train_step(
+                    (observation, time_step.step_type), state=state.icm)
```

Comment:
Instead of "if-else" here, it's better to make the interface same across the different intrinsic reward algorithms by using time_step as argument.